server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -198,10 +200,18 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
     public IndexInput openInput(String name, long fileLength, IOContext context) throws IOException {
         InputStream inputStream = null;
         try {
-            inputStream = blobContainer.readBlob(name);
-            return new RemoteIndexInput(name, downloadRateLimiter.apply(inputStream), fileLength);
+            if (context instanceof BlockIOContext) {
+                long position = ((BlockIOContext) context).getBlockStart();
+                long length = ((BlockIOContext) context).getBlockSize();
+                inputStream = blobContainer.readBlob(name, position, length);
+                byte[] bytes = downloadRateLimiter.apply(inputStream).readAllBytes();
+                return new ByteArrayIndexInput(name, bytes);
```

Comment:
ByteArrayIndexInput seems to be deprecated https://lucene.apache.org/core/7_7_2/core/org/apache/lucene/store/ByteArrayIndexInput.html

Reply:
We won't be directly using this ByteArrayIndexInput - we are just using it to download the block file on local and then we are using that local file, which will be memory mapped only. So shouldn't cause any issues